python/lib/dependabot/python/update_checker/pip_compile_version_resolver.rb

```diff
@@ -143,9 +144,13 @@ def handle_pip_compile_errors(error)
           end
 
           if error.message.match?(GIT_REFERENCE_NOT_FOUND_REGEX)
-            name = error.message.match(GIT_REFERENCE_NOT_FOUND_REGEX).
-                   named_captures.fetch("name")
-            raise GitDependencyReferenceNotFound, name
+            tag = error.message.match(GIT_REFERENCE_NOT_FOUND_REGEX).named_captures.fetch("tag")
+            constraints_section = error.message.split("Finding the best candidates:").first
+            egg_regex = /#{Regexp.escape(tag)}#egg=(\S+)/
+            name_match = constraints_section.scan(egg_regex)
+            raise GitDependencyReferenceNotFound, name_match.first.first if name_match.length == 1
```

Comment:
Do we care about the length being 1 exactly, or do we care that there is _a_ match?

Reply:
That sounds right to me. Let's add a code comment saying just that.

```suggestion
            # We can determine the name of the package from another part of the logger output if it has a unique tag name
            raise GitDependencyReferenceNotFound, name_match.first.first if name_match.length == 1
```

Reply:
Also, if we have a `raise` with a postconditional followed by another `raise`, I think the logic could be made clearer with a conventional `if / else`:

```ruby
if name_match.length == 1
   # raise with detected name
else
   # raise with synthesized name from tag
end
```

Reply:
I tried writing it this way (if/else) at first but rubocop demands it be a guard clause.